### PR TITLE
Generate coverage for Simulator based integration tests & ui tests in github actions

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR generates coverage for Simulator based integration tests in github actions workflow and uploads the report to codecov.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [x] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
I found these tests that are not included in the `ci.yml`'s test matrix.
```
broken/maint
component/maint
component/test_acquire_specified_public_ip
component/test_advancedsg_networks
component/test_affinity_groups
component/test_base_image_updation
component/test_blocker_bugs
component/test_browse_templates
component/test_browse_templates2
component/test_browse_volumes
component/test_edgezone_supportedoperations
component/test_escalations_instances
component/test_haproxy
component/test_hypervisor_capabilities
component/test_implicit_planner
component/test_interop_xd_ccp
component/test_interpod_migration
component/test_ldap
component/test_migrate_vol_to_maintained_pool
component/test_multiple_nic_support
component/test_multiple_physical_network_creation
component/test_multiple_subnets_in_isolated_network
component/test_multiple_subnets_in_isolated_network_rvr
component/test_multiple_subnets_in_vpc
component/test_multiple_subnets_in_vpc_rvr
component/test_network_mtu
component/test_organization_states
component/test_portable_ip
component/test_ps_domain_limits
component/test_ps_limits
component/test_ps_project_limits
component/test_redundant_router_cleanups
component/test_redundant_router_services
component/test_redundant_router_upgrades
component/test_resource_count_running_vms
component/test_shared_networks
component/test_storage_motion
component/test_template_access_across_domains
component/test_volume_destroy_recover
component/test_vpc_network_internal_lbrules
component/test_vpc_vms_deployment
plugins/ldap
plugins/linstor
plugins/scaleio
plugins/test_quota
smoke/test_gateway_on_shared_networks
smoke/test_guest_os
smoke/test_host_control_state
smoke/test_human_readable_logs
smoke/test_list_ids_parameter#
smoke/test_nonstrict_affinity_group
smoke/test_vm_autoscaling
testpaths/testpath_custom_disk_offering
testpaths/testpath_netmask
testpaths/testpath_queryAsyncJobResult
testpaths/testpath_stopped_vm
testpaths/testpath_usage
testpaths/testpath_uuid_event
```

**Comment by DaanHoogland** [comment]
> I found these tests that are not included in the `ci.yml`'s test matrix.

I would not include any effort on those in this PR @vishesh92 . Best would be to create a tracking issue to decide which ones to delete/fix/add.

**Review by borisstoyanov — APPROVED** [review]
LGTM

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@vishesh92 good work and proposed PR, we may need a different way to track this perhaps? All tests in smoke should be ideally run against simulator/ci.yml, but other tests arne't included as they fail or amount of effort to stablise them would be huge.

smoke/test_gateway_on_shared_networks
smoke/test_guest_os
smoke/test_host_control_state
smoke/test_human_readable_logs
smoke/test_list_ids_parameter#
smoke/test_nonstrict_affinity_group
smoke/test_vm_autoscaling

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -249,9 +256,16 @@ jobs:
...
+        run: |
+          echo "java -jar jacoco/lib/jacococli.jar report jacoco-it.exec \\" > jacoco-report.sh
+          find . | grep "target/classes" | sed 's/\/classes\//\/classes /g' | awk '{print "--classfiles", $1, "\\"}' | sort |uniq >> jacoco-report.sh
+          find . | grep "src/main/java" | sed 's/\/java\//\/java /g' | awk '{print "--sourcefiles", $1, "\\"}' | sort | uniq >> jacoco-report.sh
+          echo "--xml jacoco-coverage.xml" >> jacoco-report.sh
+
       - name: Start CloudStack Management Server with Simulator
         run: |
-          export MAVEN_OPTS="-Xmx4096m -XX:MaxPermSize=800m -Djava.security.egd=file:/dev/urandom"
+          export MAVEN_OPTS="-Xmx4096m -XX:MaxPermSize=800m -Djava.security.egd=file:/dev/urandom  -javaagent:jacoco/lib/jacocoagent.jar=address=*,port=36320,output=tcpserver"
```
Comment (rohityadavcloud) [inline .github/workflows/ci.yml:268]: perhaps we can experiment this with our smoketests against actual environment and gather/publish the jacoco file separately cc @vishesh92 @borisstoyanov @vladimirpetrov @kiranchavala @DaanHoogland - so we've a unit test coverage stats and functional/marvin test coverage

Comment (DaanHoogland) [reply .github/workflows/ci.yml:268]: I would certainly like to have them separately, but nobody dies if we don't.

**Comment by rohityadavcloud** [comment]
@vishesh92 Does this handle reports from multiple jobs and aggregate them somewhoe; could you look into (when you're free/have time): https://about.codecov.io/blog/uploading-code-coverage-in-a-separate-job-on-github-actions/

**Comment by vishesh92** [comment]
@rohityadavcloud As per my understanding, codecov groups the coverage by [flags](https://docs.codecov.com/docs/flags).

See uploaded files for this PR here: https://app.codecov.io/gh/apache/cloudstack/commit/c944865bd84a1fab86755bc164123c64291adce1

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]
LGTM - changes are in the github actions

**Pull request merged**